config.go

```diff
@@ -394,6 +416,38 @@ func (cfg *Config) unshadow() {
 	cfg.shadow = nil
 }
 
+func generateClusterSecret() ([]byte, error) {
+	encodedSecretLength := 64
+	secret := make([]byte, base64.StdEncoding.DecodedLen(encodedSecretLength))
+	_, err := crand.Read(secret)
+	if err != nil {
+		return nil, fmt.Errorf("Error reading from rand: %v", err)
+	}
+	encodedSecret := make([]byte, encodedSecretLength)
+	base64.StdEncoding.Encode(encodedSecret, secret)
+	return encodedSecret, nil
+}
+
+func clusterSecretToKey(secret []byte) (string, error) {
+	var key bytes.Buffer
+	key.WriteString("/key/swarm/psk/1.0.0/\n")
+	key.WriteString("/base64/\n")
+	key.Write(secret)
+
+	return key.String(), nil
+}
+
+func IsValidSecret(secret []byte) (bool, error) {
+	var isBase64 = regexp.MustCompile(`^[0-9A-Za-z+/]+$`).Match
```

Comment:
I think we can expose a `func ParseClusterSecret(b64secret string) ([]byte, error) {}`. You can re-use it when parsing the JSON configuration and in ipfs-cluster-service to validate user-input.